src/dashboard/ui/start.rs

```diff
@@ -10,15 +10,31 @@ use crate::dashboard::ui::utils::get_hw_overview;
 use crate::rpc_call;
 use crate::server::bootstrap::get_client_connection;
 use crate::transfer::messages::{FromClientMessage, ToClientMessage};
+use tokio::{task, time};
+use tokio::time::Duration;
+use std::cell::RefCell;
+use std::sync::Arc;
 
-pub async fn start_ui_loop(
+pub async fn start_ui_loop (
     painter: WrappedRcRefCell<DashboardPainter>,
-    event_handler: &mut DashboardEventHandler,
+    event_handler: WrappedRcRefCell<DashboardEventHandler>,
     state: WrappedRcRefCell<DashboardState>,
-    global_settings: &GlobalSettings,
+    global_settings: & GlobalSettings,
 ) -> Result<(), anyhow::Error> {
+
```

Comment:
@Kobzol How do I share the event_handler between the two parts

Reply:
Try something like this:
```rust
let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
let ticker = async move {
    let mut interval = tokio::time::interval(Duration::from_micros(500));
    loop {
        tx.send(DashboardEvent::Tick);
        interval.tick().await;
    }
};
let event_loop = async move {
    loop {
        match rx.recv().await {
            Some(dashboard_event) => {
                match dashboard_event {
                    DashboardEvent::KeyPressEvent(input) => {
                        if input == Key::Char('q') {
                            // Quits the dashboard
                            break Ok(());
                        }
                    }

                    DashboardEvent::Tick => {
                        let overview = get_hw_overview(global_settings).await?;

                        //Draw the correct dashboard ui according to the current ui state
                        match state.get().ui_state {
                            CurrentActiveUi::WorkerHwMonitorScreen => {
                                painter.get_mut().draw_dashboard_home(overview);
                            }
                        }
                    }

                    DashboardEvent::ChangeUIStateEvent(_new_state) => {
                        //todo: change what is being drawn on the dashboard by changing the ui state!
                    }
                }
            }

            None => (),
        }
    }
};
tokio::select! {
    _ = ticker => { Ok(()) }
    result = event_loop => { result }
}
```